src/dialects/ibmi/query-interface.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const { Transaction } = require('../../transaction');
+const { QueryInterface } = require('../abstract/query-interface');
+
+/**
+ Returns an object that enables the `ibmi` dialect to call underlying odbc
+ transaction functions through the connection methods instead
+
+ @class QueryInterface
+ @static
+ @private
+ */
+
+class IBMiQueryInterface extends QueryInterface {
+
+  startTransaction(transaction, options) {
+    if (!transaction || !(transaction instanceof Transaction)) {
+      throw new Error('Unable to start a transaction without transaction object!');
+    }
+
+    options = { ...options, transaction: transaction.parent || transaction };
+
+    options.transaction.name = transaction.parent ? transaction.name : undefined;
+
+    return transaction.connection.beginTransaction();
+  }
+
+  commitTransaction(transaction) {
+    if (!transaction || !(transaction instanceof Transaction)) {
+      throw new Error('Unable to commit a transaction without transaction object!');
+    }
+
+    if (transaction.parent) {
+      // Savepoints cannot be committed
```

Comment:
I think it would be better to throw when someone tries to create a savepoint instead of silently ignoring them. This is the kind of think I think users should be aware of